File: UI/vendor-panel/src/types/post.ts
export type PostType = 'PT_TEXT' | 'PT_IMAGE' | 'PT_VIDEO' | 'PT_PRODUCT_SHARE' | 'PT_JOB_POSTING' | 'PT_NEWS' | 'PT_ANNOUNCEMENT' | 'PT_POLL';
export type PostStatus = 'PS_DRAFT' | 'PS_SCHEDULED' | 'PS_ACTIVE' | 'PS_EXPIRED' | 'PS_ARCHIVED';

export interface Post {
  id: string;
  tenantId: string;
  authorId: string;
  postType: PostType;
  content?: string;
  mediaUrls: string[];
  linkedListingId?: string;
  linkedListing?: { id: string; title: string; mediaUrls: string[] };
  visibility: string;
  status: PostStatus;
  hashtags: string[];
  publishAt?: string;
  expiresAt?: string;
  pollConfig?: PollConfig;
  likeCount: number;
  commentCount: number;
  shareCount: number;
  saveCount: number;
  viewCount: number;
  createdAt: string;
  updatedAt: string;
}

export interface PollConfig {
  question: string;
  options: { text: string; votes: number }[];
  expiresAt?: string;
  multipleChoice?: boolean;
}

export interface PostComment {
  id: string;
  postId: string;
  userId: string;
  userName: string;
  content: string;
  parentId?: string;
  children?: PostComment[];
  createdAt: string;
}

export interface CreatePostDto {
  postType: PostType;
  content?: string;
  mediaUrls?: string[];
  linkedListingId?: string;
  visibility?: string;
  hashtags?: string[];
  publishAt?: string;
  expiresAt?: string;
  pollConfig?: Omit<PollConfig, 'options'> & { options: string[] };
}

File: mobile/marketplace/lib/shared/models/post.dart
class MarketplacePost {
  final String id;
  final String tenantId;
  final String authorId;
  final String postType;
  final String? content;
  final List<Map<String, String>> mediaUrls;
  final String? linkedListingId;
  final String? linkedOfferId;
  final int? rating;
  final String visibility;
  final String status;
  final DateTime? publishAt;
  final int viewCount;
  final int likeCount;
  final int commentCount;
  final int shareCount;
  final int saveCount;
  final List<String> hashtags;
  final DateTime createdAt;

  const MarketplacePost({
    required this.id,
    required this.tenantId,
    required this.authorId,
    required this.postType,
    this.content,
    required this.mediaUrls,
    this.linkedListingId,
    this.linkedOfferId,
    this.rating,
    required this.visibility,
    required this.status,
    this.publishAt,
    required this.viewCount,
    required this.likeCount,
    required this.commentCount,
    required this.shareCount,
    required this.saveCount,
    required this.hashtags,
    required this.createdAt,
  });

  factory MarketplacePost.fromJson(Map<String, dynamic> json) => MarketplacePost(
    id: json['id'] as String,
    tenantId: json['tenantId'] as String,
    authorId: json['authorId'] as String,
    postType: json['postType'] as String,
    content: json['content'] as String?,
    mediaUrls: (json['mediaUrls'] as List<dynamic>?)
        ?.map((e) => Map<String, String>.from(e as Map))
        .toList() ?? [],
    linkedListingId: json['linkedListingId'] as String?,
    linkedOfferId: json['linkedOfferId'] as String?,
    rating: json['rating'] as int?,
    visibility: json['visibility'] as String? ?? 'PUBLIC',
    status: json['status'] as String? ?? 'DRAFT',
    publishAt: json['publishAt'] != null ? DateTime.parse(json['publishAt'] as String) : null,
    viewCount: json['viewCount'] as int? ?? 0,
    likeCount: json['likeCount'] as int? ?? 0,
    commentCount: json['commentCount'] as int? ?? 0,
    shareCount: json['shareCount'] as int? ?? 0,
    saveCount: json['saveCount'] as int? ?? 0,
    hashtags: List<String>.from(json['hashtags'] as List? ?? []),
    createdAt: DateTime.parse(json['createdAt'] as String),
  );
}
